pkg/compact/compact.go

```diff
@@ -913,22 +913,42 @@ func (c *BucketCompactor) Compact(ctx context.Context) error {
 			return errors.Wrap(err, "build compaction groups")
 		}
 		finishedAllGroups := true
+		var wg sync.WaitGroup
+		errChan := make(chan error, len(groups))
```

Comment:
I think we need something more similiar to this https://github.com/improbable-eng/thanos/pull/887/files still.
This is for consistency and readability reasons. so:

1. Run `concurrency` number of go routines BEFORE ` for{ ` loop (891 line)
2. Do the whole sync Iterate over groups
```
   for _, g := range groups {
````. 
3. Send `g` via some channel with 0 queue.
4. In go routines do the g (`g.Compact`) and mark `finishedAllGroups` to false if there are still some group to run. Make sure to make it `atomic.Value` or use `sync.Mutex`. 
5. After ` for _, g := range groups {` close channel and wait for all groups to finish. If `finishedAllGroups == true` return otherwise continue